linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2523,26 +2496,23 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
                 Transient(ObjectClassName(expr))))
 
           case _ =>
-            defaultApply(StringClassType)
+            default
```

Comment:
Same here.

Perhaps it would be safer to remove `newReceiver` and `newArgs`, and recompute what's necessary in the few remaining cases that need them?